File: frontend/src/app/models/maestro.ts
import { IUsuario } from "./usuario"

export interface ITipo {
    IdTipo: number
    Tipo: string
}

export interface IMaestro extends IUsuario{

    NoEmpleado: number,
    Tipo: ITipo

}

export class Maestro implements IMaestro{
    NoEmpleado: number = 0
    Tipo: ITipo = { IdTipo: 0, Tipo: "" }
    Nombre: string = ""
    ApellidoPaterno: string = ""
    ApellidoMaterno: string = ""
    Usuario: string = ""
    Estatus: string = ""
    Contrasena?: string | undefined = ""

}

File: backend/src/main/java/models/Maestro.java
package models;

public class Maestro extends Usuario{
	
	private int NoEmpleado;
	private Tipo Tipo;
	
	public int getNoEmpleado() {
		return NoEmpleado;
	}
	
	public void setNoEmpleado(int noEmpleado) {
		NoEmpleado = noEmpleado;
	}
	
	public Tipo getTipo() {
		return Tipo;
	}
	
	public void setTipo(Tipo tipo) {
		this.Tipo = tipo;
	}
	
}
